gfx.cpp

```diff
@@ -6812,11 +6812,14 @@ class GfxInternal
                             }
                         }
                     }
+                    transitionResource(sbt_buffer, D3D12_RESOURCE_STATE_COPY_DEST);
+                    submitPipelineBarriers();
```

Comment:
Issue with this is that we know have this `submitPipelineBarriers()` happening in a loop, some minimal batching would be good (like pre-transition all SBT buffers before the copies, then transition back after the copies, rather than one by one like this).

Reply:
Good point, I was initially thinking that if we auto handle implicit transitions then this would be a noop. But ive updated the PR to batch the transitions anyway